src/sagemaker/tensorflow/model.py

```diff
@@ -83,11 +83,18 @@ def prepare_container_def(self, instance_type):
         """
         deploy_image = self.image
         if not deploy_image:
-            region_name = self.sagemaker_session.boto_session.region_name
+            region_name = self.sagemaker_session.region_name
             deploy_image = create_image_uri(region_name, self.__framework_name__, instance_type,
                                             self.framework_version, self.py_version)
+
         deploy_key_prefix = self.key_prefix or self.name or name_from_image(deploy_image)
-        self._upload_code(deploy_key_prefix)
+
+        no_internet = get_config_value('local.no_internet', self.sagemaker_session.config)
+        print('no_internet: %s  local_mode: %s' % (no_internet, self.sagemaker_session.local_mode))
+        if self.sagemaker_session.local_mode and no_internet:
+            self.uploaded_code = None
```

Comment:
this is a typo - should be self_uploaded_code, but you could probably remove this entire block anyway